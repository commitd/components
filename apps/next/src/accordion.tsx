import {
  Accordion,
  AccordionContent,
  AccordionHeader,
  AccordionItem,
} from '@committed/ds'

export const AccordionExample = () => (
  <Accordion type="multiple">
    <AccordionItem value="item-1">
      <AccordionHeader>Item 1</AccordionHeader>
      <AccordionContent>
        Here goes the content for the accordion item 1.
      </AccordionContent>
    </AccordionItem>
    <AccordionItem value="item-2">
      <AccordionHeader>Item 2</AccordionHeader>
      <AccordionContent>
        Here goes the content for the accordion item 2.
      </AccordionContent>
    </AccordionItem>
    <AccordionItem value="item-3">
      <AccordionHeader>Item 3</AccordionHeader>
      <AccordionContent>
        Here goes the content for the accordion item 3.
      </AccordionContent>
    </AccordionItem>
  </Accordion>
)
