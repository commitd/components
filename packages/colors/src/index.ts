export { presetColors } from './preset'
export type { Color } from './preset'
