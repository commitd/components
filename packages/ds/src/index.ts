// component exports
export * from './components'
export * from './utils'

// reexports from ss
export { css, styled } from '@committed/ds-ss'
