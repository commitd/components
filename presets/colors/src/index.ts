export {
  neutralColors,
  preset as presetColors,
  primaryColors,
  semanticColors,
} from './preset'
export type { Color, Prefix, SemanticColor } from './preset'
