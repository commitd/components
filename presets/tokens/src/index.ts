export { preset as presetTokens } from './preset'
