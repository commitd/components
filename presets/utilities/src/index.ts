export { preset as presetUtilities, surfaces } from './preset'
export type { Surface, SurfaceVariants } from './preset'

