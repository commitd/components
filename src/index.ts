export * from './components'
export * from './theme'
export * from './hooks'

// Reexport
export { styled } from '@material-ui/styles'

// export shortcodes for convenience
import * as shortcodes from './components'
export { shortcodes }
