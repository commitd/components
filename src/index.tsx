export * from './components'
export * from './theme'
export * from './styles'
export * from './util'
// Some explicit exports are required to resolve ambiguity
export { ThemeProvider, colors } from './theme'
export { makeStyles, styled, useTheme, withTheme } from './styles'
export type { ThemeProviderProps } from './theme'
export type { ComponentCreator, WithTheme } from './styles'
