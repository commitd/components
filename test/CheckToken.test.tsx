import React from 'react'
import { renderLight, renderDark, userEvent } from './setupTests'
import { CheckToken } from '../src'

it('renders light without crashing', () => {
  const { asFragment } = renderLight(<CheckToken />)
  expect(asFragment()).toMatchSnapshot()
})

it('renders dark without crashing', () => {
  const { asFragment } = renderDark(<CheckToken />)
  expect(asFragment()).toMatchSnapshot()
})

it('clicking calls onClick', () => {
  const text = 'Test'
  const onClick = jest.fn()
  const { getByLabelText, getByText } = renderDark(
    <CheckToken aria-label="check-token" onClick={onClick}>
      {text}
    </CheckToken>
  )
  userEvent.click(getByLabelText('check-token'))
  userEvent.click(getByText(text))
  expect(onClick).toBeCalledTimes(2)
})

it('element can be controlled: true', () => {
  const text = 'Test'
  const onClick = jest.fn()
  const { getByRole } = renderDark(
    <CheckToken selected={true} onClick={onClick}>
      {text}
    </CheckToken>
  )
  userEvent.click(getByRole('button', { pressed: true }))
  expect(onClick).toBeCalledTimes(1)
})

it('element can be controlled: false ', () => {
  const text = 'Test'
  const onClick = jest.fn()
  const { getByRole } = renderDark(
    <CheckToken selected={false} onClick={onClick}>
      {text}
    </CheckToken>
  )
  userEvent.click(getByRole('button', { pressed: false }))
  expect(onClick).toBeCalledTimes(1)
})

it('value is passed to handler ', () => {
  const text = 'Test'
  const value = 'Testing'
  const onClick = jest.fn()
  const { getByRole } = renderDark(
    <CheckToken value={value} onClick={onClick}>
      {text}
    </CheckToken>
  )
  userEvent.click(getByRole('button', { pressed: false }))
  expect(onClick).toBeCalledWith(expect.anything(), value)
})
