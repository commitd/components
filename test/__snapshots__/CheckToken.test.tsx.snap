// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders dark without crashing 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiToggleButton-root makeStyles-root ForwardRef(withPositioning)-root ForwardRef(withPositioning)-root"
    tabindex="0"
    type="button"
    value="check"
  >
    <span
      class="MuiToggleButton-label"
    >
      <div
        class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
      </div>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;

exports[`renders light without crashing 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiToggleButton-root makeStyles-root ForwardRef(withPositioning)-root ForwardRef(withPositioning)-root"
    tabindex="0"
    type="button"
    value="check"
  >
    <span
      class="MuiToggleButton-label"
    >
      <div
        class="MuiBox-root MuiBox-root Styled(MuiBox)-root Styled(MuiBox)-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
          />
        </svg>
      </div>
    </span>
    <span
      class="MuiTouchRipple-root"
    />
  </button>
</DocumentFragment>
`;
